python/paddle/fluid/tests/unittests/ipu/test_dy2static_fp16_ipu.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import numpy as np
+import unittest
+import sys
+import os
+import paddle
+import paddle.fluid as fluid
+from paddle.jit import to_static
+from paddle.utils.cpp_extension import load
+from paddle.optimizer.lr import LRScheduler
+import shutil
+
+SEED = 2022
+
+
+class SimpleLayer(paddle.nn.Layer):
+
+    def __init__(self, use_ipu=False):
+        super(SimpleLayer, self).__init__()
+        self.use_ipu = use_ipu
+        self.conv = paddle.nn.Conv2D(in_channels=3,
+                                     out_channels=1,
+                                     kernel_size=2,
+                                     stride=1)
+
+    def forward(self, x, target=None):
+        x = self.conv(x)
+        x = paddle.fluid.layers.flatten(x, axis=1)
+        if target is not None:
+            x = paddle.fluid.layers.softmax(x)
+            loss = paddle.fluid.layers.cross_entropy(x, target)
+            if self.use_ipu:
+                loss = paddle.fluid.layers.identity_loss(loss, 1)
+            else:
+                loss = paddle.mean(loss)
+            return x, loss
+        return x
+
+
+class TestBase(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        paddle.disable_static()
+        cls.save_path = os.path.join(os.getcwd(), "temp_save")
```

Comment:
Recommend to use `tempfile`,for example ( see test_grad.py)
```python
import tempfile

class TestGradLinear(TestGrad):

    def setUp(self):

        self.temp_dir = tempfile.TemporaryDirectory()
        self.infer_model_path = os.path.join(self.temp_dir.name,
                                             'double_grad_infer_model')
        self.train_model_path = os.path.join(self.temp_dir.name,
                                             'double_grad_train_model')

    def tearDown(self):
        self.temp_dir.cleanup()
```

Reply:
Updated.